sgx/epid-types-sys/build.rs

```diff
@@ -2,16 +2,60 @@
 
 //! Build script to generate bindings for the Intel SGX SDK core FFI types
 
-use bindgen::Builder;
+use bindgen::{
+    callbacks::{IntKind, ParseCallbacks},
+    Builder,
+};
 use mcbuild_utils::Environment;
 
+#[derive(Debug)]
+struct Callbacks;
+
+impl ParseCallbacks for Callbacks {
+    fn int_macro(&self, name: &str, _value: i64) -> Option<IntKind> {
+        if name.ends_with("_SIZE") || name.ends_with("_BYTES") {
+            Some(IntKind::Custom {
+                name: "usize",
+                is_signed: false,
+            })
+        } else {
+            None
+        }
+    }
+
+    fn item_name(&self, name: &str) -> Option<String> {
+        if name.starts_with("_sgx_") {
+            Some(name[1..].to_owned())
+        } else if name.starts_with('_') {
+            let mut retval = "sgx".to_owned();
+            retval.push_str(name);
+            // Cleanup the _quote_nonce, _platform_info, and _update_info_bit types.
+            //
+            // Yelling in the wilderness:
+            //
+            // Is there someone going around teaching students that "`_t` means it's a C type"
+            // without further explanation? The `_t` suffix exists for POSIX to namespace itself
+            // from the code of mere mortals, not for every damned fool to remind themselves what
+            // language they are writing code for.
+            //   -jmc
```

Comment:
hahaha...

My recollection from C++ is that it is common that someone will put `_t` on a type, and no `_t` for variables of that type. Because in both cases they are typically snake case, so it can make it a little easier to humans to parse it at a glance. At least that was what I understood about it. I think it's common in C standard library also with like the `cstdint.h` header having `uint64_t` and `size_t` and `usize_t`.

Reply:
Oh yeah, I've seen it all over, my complaint is that it conflicts with http://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html ;-)